src/cljs/athens/electron.cljs

```diff
@@ -214,161 +216,122 @@
 
   (defn get-db-name
     "From a dbpath find out db name.
-     Args : dbpath : type str
      e.g : /home/sid597/Desktop/athens db/test db/index.transit --> test db"
     [dbpath]
-    (let  [split-path (clojure.string/split dbpath #"/")]
-      (nth  split-path (- (count split-path) 2))))
+    (.basename path (.dirname path dbpath)))
 
   (defn check-duplicate-db
     "Check if a db is already in the list.
-     Args :
-      db-list    : type vector : A list of all the current
-                                 dbs in db-picker list.
-      check-path : type str    : Path which is to be checked.
      Implementation thoughts
-      - Should this be done whenever a new db is added or 
-        should we maintain a set universally and check that?
-      - No performance issues if this is done on every new db 
-        addition because not many dbs would be made frequently
-        to cause performance issue. What about merging 2 large
-        dbs? Not there yet, but in that case also calculating
+      - Should this be done whenever a new db is added or should we maintain a
+        set universally and check that?
+      - No performance issues if this is done on every new db addition because
+        not many dbs would be made frequently to cause performance issue. What
+        about merging 2 large dbs? Not there yet, but in that case also calculating
         the set of all paths is O(n).
-      - Also re-frame is in-memory so saving data which can be
-        easily calculated is not a good idea I think"
-
+      - Also re-frame is in-memory so saving data which can be easily calculated is
+        not a good idea I think"
     [db-list check-path]
-    (let [path-set (into #{}
-                         (reduce
-                           (fn [paths db-list-item]
-                             (conj paths (get db-list-item :path)))
-                           [] db-list))]
-      (contains? path-set check-path)))
+    (contains? (into #{} (map #(:path %) db-list)) check-path))
```

Comment:
Do you know if using `set` is better than `into #{}`? Genuine question, I don't know the difference.

Reply:
Yeah I also don't know the difference, I remembered using `into #{}` to solve 4clojure problems so here also I used it. From what I could find they both have the [same performance](https://insideclojure.org/2015/01/06/set-perf/) as of Clojure 1.7

Reply:
```clojure
(into #{} [1 2 1])
;=> #{1 2}

(set [1 2 1])
;=> #{1 2}

;; Though I expected it to behave like `clojure.core/sorted-set`
(apply sorted-set [1 2 1])
;=> #{1 2}
(sorted-set [1 2 1])
;=> #{[1 2 1]}
```

Would this be more readable:
```clojure
(seq (filter #(= check-path (:path %)) db-list))
```

Reply:
> `(seq (filter #(= check-path (:path %)) db-list))` 

This looks better and readable, also learned `seq` is the idiomatic way of checking if a seq is empty or not.